### PR TITLE
k8s:  Add a dev only feature flag and simple service to get a client

### PR DESCRIPTION
**Pull request by ryantxu**
This is a basic PR to add a new dev-only feature flag that will wire in a kubernetes client.

This will first try:
1.  `rest.InClusterConfig()`
2.  RESTConfigFromKubeConfig, from `os.ReadFile(filepath.Join(os.UserHomeDir(), ".kube", "config"))`

When running as an admin with localhost and having started `minikube`, this now shows:
http://localhost:3000/admin/storage/k8s

image

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ replace github.com/denisenkom/go-mssqldb => github.com/grafana/go-mssqldb v0.0.0
 
 // Override k8s.io/client-go outdated dependency, which is an indirect dependency of grafana/loki.
 // It's also present on grafana/loki's go.mod so we'll need till it gets updated.
-replace k8s.io/client-go => k8s.io/client-go v0.22.1
+replace k8s.io/client-go => k8s.io/client-go v0.25.0
```
Comment (ryantxu) [inline go.mod:12]: @gabor -- is this update OK?  loki is now using v25

File: pkg/services/store/k8saccess/service.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+type k8sAccess struct {
+	apihelper *httpHelper
+	sysclient *kubernetes.Clientset
+	syserr    error
+}
+
+func ProvideK8SAccess(toggles featuremgmt.FeatureToggles, router routing.RouteRegister) K8SAccess {
+	access := &k8sAccess{}
+
+	if !toggles.IsEnabled(featuremgmt.FlagK8s) {
```
Comment (jalevin) [inline pkg/services/store/k8saccess/service.go:30]: Might be good to throw a comment here that explains the routes don't get mounted. I don't think that's obvious

**Review by jalevin — APPROVED** [review]
Nice, I like making the kubernetes config it's own service.

File: public/app/features/storage/k8s/K8SPage.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const info = useAsync(() => {
+    return getBackendSrv().get('/api/k8s/info');
+  });
+
+  const renderView = () => {
+    if (info.value) {
+      return <pre>{JSON.stringify(info.value, null, 2)}</pre>;
+    }
+
+    return <div className={styles.wrapper}>K8s FTW</div>;
```
Comment (toddtreece) [inline public/app/features/storage/k8s/K8SPage.tsx:34]: tested locally, and i got it working eventually. it wasn't clear if this was the default message, a success message, or a failure

Comment (ryantxu) [reply public/app/features/storage/k8s/K8SPage.tsx]: Updated so we now show this:  
image

can still use work, but at least it says something is wrong :)

File: pkg/services/store/k8saccess/http.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		}
+		return response.JSON(200, info)
+	}
+	return response.JSON(500, map[string]interface{}{
+		"error": "no client initalized",
+	})
+}
+
+func (s *httpHelper) doProxy(c *models.ReqContext) {
+	if s.access.sys != nil {
```
Comment (toddtreece) [inline pkg/services/store/k8saccess/http.go:42]: should `sys.err` be checked here?

**Review by toddtreece — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (gabor) [reply go.mod:12]: @ryantxu hmm.. actually, i'm not sure if we use loki-code in grafana anymore. when https://github.com/grafana/grafana/commit/da74dba7c8d3da59d4187ae9414c7257985f9e48 was committed, we switched to the streaming-json-parser, and stopped using the loki version. 

(but i'm not sure what that lines does exactly though)

**Pull request merged**